packages/arb-token-bridge-ui/src/components/common/NetworkSelectionContainer.tsx

```diff
@@ -67,19 +84,49 @@ export const NetworkSelectionContainer = ({
           | undefined
       ) => void
     ) => {
-      switchNetwork?.(Number(chainId))
+      setSelectedChainId(chainId)
+      if (isConnected) {
+        switchNetwork?.(Number(chainId))
+      } else {
+        // this is to make sure it's run after `setSelectedChainId,
+        // otherwise there'll be a race condition where the previous chain is used on reload
+        setTimeout(() => window.location.reload(), 0)
+      }
       close?.() //close the popover after option-click
     },
-    [switchNetwork]
+    [isConnected, switchNetwork]
   )
 
+  useEffect(() => {
+    if (isConnected && chain.id !== selectedChainId) {
+      // when user is connected, use wallet's chain
+      setSelectedChainId(chain.id)
+    }
+    // When user was connected to an unsupported chain and disconnected
+    // set mainnet as selected chain
+    if (!isConnected && !walletConnectChain) {
+      setSelectedChainId(ChainId.Mainnet)
+    }
+    setWalletConnectChain(selectedChainId)
```

Comment:
Maybe we can do something like:

```
setWalletConnectChain(!isConnected ? selectedChainId : undefined)
```

This only makes sense to be added to QP if we aren't connected. Otherwise it's just a bit confusing as a user if I'm connected to metamask but I see a wallet connect parameter

Reply:
was thinking along the line where we want to state the connected chain explicitly with the query param so user could enter with it as "from" (might move it to "from" in the future)

Reply:
@spsjvc ser this is related to your refactoring work so your input is super valuable hehe